commit bd8f56c56157b7dc5b9007835b6f9d124ecb9eaf
Author: Jason King <jason.king@joyent.com>
Date:   2019-07-15T14:20:10+00:00 (3 months ago)
    
    OS-7860 Install libsunw_crypto.a to proto area during build
    Reviewed by: John Levon <john.levon@joyent.com>
    Reviewed by: Mike Gerdts <mike.gerdts@joyent.com>
    Approved by: Mike Gerdts <mike.gerdts@joyent.com>

diff --git a/openssl1x/install-sfw-64 b/openssl1x/install-sfw-64
index f619d2d..d33d1df 100644
--- a/openssl1x/install-sfw-64
+++ b/openssl1x/install-sfw-64
@@ -24,7 +24,7 @@
 # Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
 # Use is subject to license terms.
 #
-# Copyright (c) 2010 Joyent Inc.
+# Copyright 2019 Joyent, Inc.
 #
 
 set -o errexit
@@ -67,4 +67,13 @@ _install L ../../../lib/amd64/libsunw_ssl.so.${LIBVER} \
     ${USRLIBDIR}/libsunw_ssl.so.${LIBVER}
 _install L ../../../lib/amd64/libsunw_ssl.so ${USRLIBDIR}/libsunw_ssl.so
 
+# The pivy tools as well as kbmd deliberately statically
+# link against libsunw_crypto.a to prevent sharing of code pages
+# to mitigate against certain timing attacks.  We install the .a
+# into the proto area in a location that won't be seen (and
+# unintentionally used) by other software, but do not
+# deliver the static library in the platform image.
+mkdir -p ${DESTDIR}/.build
+_install D libcrypto.a ${DESTDIR}/.build/libsunw_crypto.a 0755
+
 exit 0
